src/create.ts

```diff
@@ -1,12 +1,171 @@
-import { DIDDocument } from "did-resolver";
-import { Authority, ConfigOptions } from "./types";
+import { Resolver, DIDDocument } from 'did-resolver';
+import {
+  getResolver,
+  eosioChainRegistry as defaultChainRegistry,
+  REGEX_ACCOUNT_NAME,
+  REGEX_CHAIN_ID,
+  REGEX_CHAIN_NAME 
+} from 'eosio-did-resolver';
+import { Authority, ConfigOptions } from './types';
+import fetch, { FetchError } from 'node-fetch';
+import { JsonRpc, Api } from 'eosjs';
+import { SignatureProvider } from 'eosjs/dist/eosjs-api-interfaces';
+import { TextEncoder, TextDecoder } from 'util';
+
+const resolver = new Resolver(getResolver());
 
 export default async function create(
-    creator: string,
-    name: string,
-    owner: Authority,
-    active: Authority,
-    options?: ConfigOptions): Promise<DIDDocument> {
-
-    return { id: 'did:eosio:stub' };
-}
\ No newline at end of file
+  chain: string,
+  creator: string,
+  name: string,
+  owner: Authority,
+  active: Authority,
+  signatureProvider: SignatureProvider,
+  options?: ConfigOptions
+): Promise<DIDDocument> {
+  // the user may want to override these values.
+  const additionalOptions = {
+    receiverAccount: options?.receiverAccount ? options.receiverAccount : 'eosio',
+    authorization: [
+      {
+        actor: creator,
+        permission: options?.creatorPermission ? options.creatorPermission : 'active',
+      },
+    ],
+    buyrambytes: {
+      bytes: options?.buyrambytes ? options.buyrambytes : 8192,
+    },
+    delegatebw: {
+      stakeNetQuantity: options?.stakeNetQuantity ? options.stakeNetQuantity : '1.0000 EOS',
+      stakeCpuQuantity: options?.stakeCpuQuantity ? options.stakeCpuQuantity : '1.0000 EOS',
+      transfer: false,
+    },
+    config: {
+      blocksBehind: 3,
+      expireSeconds: 30,
+    },
+  };
+
+  validateAccountName(creator);
+  validateAccountName(name);
+
+  const chainRegistry = {
+    ...defaultChainRegistry,
+    ...options?.registry,
+  };
+
+  const chainData = getChainData(chainRegistry, chain);
+
+  let fetchAttempts = 0;
+  for (const service of chainData.service) {
+    fetchAttempts++;
+
+    const rpc = new JsonRpc(service.serviceEndpoint, { fetch });
```

Comment:
this is using the `node-fetch` package which isn't necessary for browsers. I suggest we make this part of the ConfigOptions